skyfield/data/mpc.py

```diff
@@ -226,6 +227,31 @@ def comet_orbit(row, ts, gm_km3_s2):
     comet._rotation = inertial_frames['ECLIPJ2000'].T
     return comet
 
+def comet_orbits(rows, ts, gm_km3_s2):
+    p = np.empty(len(rows))
+    e = rows.eccentricity.values
+    parabolic = (e == 1.0)
```

Comment:
This will be so rare that I think you could elide a step here: instead of starting with `p` empty, and then creating an array almost the size of `p` with `a * (1.0 - (e[~parabolic])**2)` only to then copy almost all those values into `p`, try just starting with `p = a * (1.0 - e**2)` which will provide you with a `p` that is mostly or entirely already populated. Then, use the `parabolic` mask to overwrite those few elements of `p` that will now be zero.

Reply:
I might be missing something here, `a` is calculated from perihelion distance and eccentricity; if we go ahead and calculate `a` for all comets it would cause divide by 0 for any comet with e=1.0.

Reply:
maybe `p` could be initialized as `1.0 - e**2` and then later multiplied by `a` for non-parabolic cases

Reply:
but then we're still using a boolean mask that's almost but not quite full, and possibly reducing readability. Do you think that would still be an improvement?

Reply:
It's actually not too bad after I tried it, I went ahead and committed what I came up with.